include/devicetree.h

```diff
@@ -1316,6 +1316,22 @@
 #define DT_ANY_INST_ON_BUS(bus) \
 	(UTIL_LISTIFY(DT_NUM_INST(DT_DRV_COMPAT), DT_INST_ON_BUS_OR, bus) 0)
 
+/**
+ * @def DT_INST_FOREACH
+ *
+ * @brief Call specified macro for all nodes with compatible DT_DRV_COMPAT
+ *
+ * @details This macro will scan for all DT_INST_ device nodes for that driver.
+ * The macro then calls the supplied macro with the instance number.
+ * This macro can be used for example to call the init macro of a driver
+ * for each device specified in the device tree.
+ *
+ * @param inst_expr Macro or function that is called for each device node.
+ * Has to accept instance_number as only parameter.
+ */
+#define DT_INST_FOREACH(inst_expr) \
+	UTIL_LISTIFY(DT_NUM_INST(DT_DRV_COMPAT), DT_CALL_WITH_ARG, inst_expr)
```

Comment:
Just checking -- this does the right thing if DT_NUM_INST(DT_DRV_COMPAT) is 0, right?

Reply:
Uses LISTIFY internally, and that uses REPEAT and REPEAT checks wether the variable is 0 in the beginning so should be fine. I just tested it though to be sure and it looks good.

Reply:
Thanks for confirming.